### PR TITLE
Track tabChanged and authorFilterSelectionChanged events in post list

### LINKED ISSUE #9651: Post List: Track tab changes and author filter changes
Add tracking for
- onTabChanged - `post_list_author_filter_changed | site info, author_filter_selection = "ME/EVERYONE"`
- onAuthorFilterSelectionChanged - `post_list_tab_changed | site info, selected_tab = "PUBLISHED, DRAFTS, SCHEDULED, TRASHED`

### PR DESCRIPTION
**Pull request by malinajirka**
Fixes #9651 

Adds tracking of two events on Post List screen
- on tab changed
- on author filter changed

I've also added an unrelated comment to the WPMainActivity as it felt unnecessary to create a separate PR for it. Feel free to update the comment.

To test:
1. My Site
2. Blog Posts
3. Swipe to Drafts (post_list_tab_changed)
4. Click on Published (post_list_tab_changed)
5. Change author selection filter to me (post_list_author_filter_changed)
6. Change author selection filter to Everyone (post_list_author_filter_changed)
7. Rotate your device (nothing should be logged)
7. Check the log cat and make sure the events are being logged correctly 

Merge instructions:
~- mark this PR as ready for review~
- merge the PR

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PostListMainViewModel.kt [hunk 1]
```diff
@@ -238,6 +241,12 @@ class PostListMainViewModel @Inject constructor(
     fun onTabChanged(position: Int) {
         val currentPage = POST_LIST_PAGES[position]
         updateViewStateTriggerPagerChange(isFabVisible = FAB_VISIBLE_POST_LIST_PAGES.contains(currentPage))
+
+        AnalyticsUtils.trackWithSiteDetails(
+                POST_LIST_TAB_CHANGED,
+                site,
+                mutableMapOf("selected_tab" to currentPage.toString() as Any)
```
Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PostListMainViewModel.kt]: I think we can simplify this a bit like so:

```suggestion
                mapOf("selected_tab" to currentPage.toString())
```

I'd also suggest moving the "selected_tab" and the "author_filter_selection" to some kind of constant.

Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/PostListMainViewModel.kt]: Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/9654/commits/7c5243c4a8977df86824dd4ee704cf77e75ba2e7

**Comment by malinajirka** [comment]
Thanks @oguzkocer! It's ready for another round.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PostListMainViewModel.kt (hunk 1 above)
Comment (oguzkocer) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/PostListMainViewModel.kt]: @malinajirka It looks like `as Any` conversions are still there 😢

Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/PostListMainViewModel.kt]: Sorry 😞 , fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/9654/commits/22cc80c6847192ea3b4073e81ad303db7c919f47

**Pull request merged**